### PR TITLE
Support pytorch 2.0

### PR DESCRIPTION
**Pull request by take-cheeze**
closes https://github.com/pfnet/pytorch-pfn-extras/issues/652

- [x] Fix lint: mypy
  - Update mypy to `0.991`
  - Insert `type: ignore` comment to existing mypy errors found by update
  - Update patched `torch/_C/__init__.pyi`

### CODE REVIEW

File: pytorch_pfn_extras/onnx/pfto_exporter/export.py [hunk 1]
```diff
@@ -124,17 +124,17 @@ def onnx_node_doc_string(onnx_node: torch._C.Node, torch_node: torch._C.Node) ->
...
-    torch.int8: onnx.TensorProto.DataType.INT8,
-    torch.int16: onnx.TensorProto.DataType.INT16,
-    torch.int32: onnx.TensorProto.DataType.INT32,
-    torch.int64: onnx.TensorProto.DataType.INT64,
-    torch.bool: onnx.TensorProto.DataType.BOOL,
-    torch.float64: onnx.TensorProto.DataType.DOUBLE,
-    torch.float16: onnx.TensorProto.DataType.FLOAT16,
-    torch.complex64: onnx.TensorProto.DataType.COMPLEX64,
-    torch.complex128: onnx.TensorProto.DataType.COMPLEX128,
+    torch.float32: onnx.TensorProto.DataType.FLOAT,  # type: ignore[attr-defined]
```
Comment (take-cheeze) [inline pytorch_pfn_extras/onnx/pfto_exporter/export.py:127]: I wish this was supported in onnx... https://github.com/protocolbuffers/protobuf/issues/8175

File: pytorch_pfn_extras/training/manager.py [hunk 2]
```diff
@@ -463,8 +463,7 @@ def _finalize_extensions(self) -> None:
             # Some mock objects for tests give errors
             # if we use `getattr`
             try:
-                if entry.extension.finalize:
-                    entry.extension.finalize(self)
+                entry.extension.finalize(self)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/manager.py:466]: Is this fix safe?

**Review by kmaehashi — COMMENTED** [review]
LGTM except for mionr comment!

File: pytorch_pfn_extras/training/manager.py (hunk 2 above)
Comment (take-cheeze) [reply pytorch_pfn_extras/training/manager.py:466]: https://github.com/pfnet/pytorch-pfn-extras/blob/e25f1a8b87397e94fdd668d049bb85d39454008b/pytorch_pfn_extras/training/extension.py#L188 seems not safe so will fix

**Review by kmaehashi — APPROVED** [review]
LGTM!

**Pull request merged**